### PR TITLE
add ability to set GCName via runtimeconfig.

### PR DESCRIPTION
**Pull request by mangod9**
gcname was only enabled via `DOTNET_GCName`, this change enables setting it via `System.GC.GCName` in runtimeconfig.json.

### CODE REVIEW

File: src/coreclr/vm/gcheaputilities.cpp [hunk 1]
```diff
@@ -334,8 +335,7 @@ HRESULT GCHeapUtilities::LoadAndInitialize()
     assert(g_gc_load_status == GC_LOAD_STATUS_BEFORE_START);
     g_gc_load_status = GC_LOAD_STATUS_START;
 
-    LPWSTR standaloneGcLocation = nullptr;
-    CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_GCName, &standaloneGcLocation);
+    LPCWSTR standaloneGcLocation = Configuration::GetKnobStringValue(W("System.GC.GCName"), CLRConfig::EXTERNAL_GCName);
```
Comment (mangod9) [inline src/coreclr/vm/gcheaputilities.cpp:338]: was debating between `System.GC.Name` and `System.GC.GCName`, but thought it should be consistent with `DOTNET_GCName`?

**Comment by cshung** [comment]
@mrsharm, shall we update the behavior of the `GetConfigurationVariables` API?

File: src/coreclr/vm/gcheaputilities.cpp (hunk 1 above)
Comment (Maoni0) [reply src/coreclr/vm/gcheaputilities.cpp:338]: generally we drop the `GC` part in the runtime config since it already starts with `System.GC`.

Comment (mangod9) [reply src/coreclr/vm/gcheaputilities.cpp:338]: ok will change after the CI finishes.

**Comment by mrsharm** [comment]
> @mrsharm, shall we update the behavior of the `GetConfigurationVariables` API?

Sure - will work on this.

File: src/coreclr/vm/gcheaputilities.cpp (hunk 1 above)
Comment (mangod9) [reply src/coreclr/vm/gcheaputilities.cpp:338]: there is one CI failure in the `monointerpreter` macOS leg which is due to this change. Dont believe the change should be affecting the scenario, but the test infra is most likely failing due to it (perhaps they are using GCName or something.

Comment (mangod9) [reply src/coreclr/vm/gcheaputilities.cpp]: updated the name to `System.GC.Name`

**Comment by mangod9** [comment]
There is a CI issue which is blocking re-running the wasm leg. It had passed on the previous run so this should be good to merge. Please CR.

**Review by mrsharm — APPROVED** [review]
LGTM - I'll be following this PR up with updating our GC Configs so that this configuration shows up in our ``GetConfigurationVariables`` API.

**Comment by mangod9** [comment]
> LGTM - I'll be following this PR up with updating our GC Configs so that this configuration shows up in our `GetConfigurationVariables` API.

I plan to backport this to 7, would the API change require porting to 7 as well?

**Comment by mrsharm** [comment]
> > LGTM - I'll be following this PR up with updating our GC Configs so that this configuration shows up in our `GetConfigurationVariables` API.
> 
> I plan to backport this to 7, would the API change require porting to 7 as well?

Since the change will be a trivial and risk-free one, I think we could add the changes to gcconfig.h in one PR i.e. for the backport, merge the changes from this PR and the one I plan to create (which should be a few lines only).

**Comment by mrsharm** [comment]
Added the new config to ``gcconfig.h`` and now the ``GetConfigurationVariables`` call output:

```
{ ServerGC : False }
{ ConcurrentGC : True }
{ RetainVM : False }
{ NoAffinitize : False }
{ GCCpuGroup : False }
{ GCLargePages : False }
{ HeapCount : 1 }
{ GCHeapAffinitizeMask : 0 }
{ GCHeapAffinitizeRanges :  }
{ GCHighMemPercent : 0 }
{ GCHeapHardLimit : 0 }
{ GCHeapHardLimitPercent : 0 }
{ GCHeapHardLimitSOH : 0 }
{ GCHeapHardLimitLOH : 0 }
{ GCHeapHardLimitPOH : 0 }
{ GCHeapHardLimitSOHPercent : 0 }
{ GCHeapHardLimitLOHPercent : 0 }
{ GCHeapHardLimitPOHPercent : 0 }
{ GCConserveMem : 0 }
{ GCName :  }
```

**Review by cshung — APPROVED** [review]

**Pull request merged**